include/zeek-compat.h

```diff
@@ -200,6 +210,42 @@ using ::attr_tag::ATTR_OPTIONAL;
 
 //// Wrapper functions for functionality that differs by version.
 
+namespace spicy::zeek::compat {
+#if ZEEK_VERSION_NUMBER < 40200 // Zeek < 4.2
+using AnalyzerTag = ::zeek::analyzer::Tag;
+using FileAnalysisTag = ::zeek::file_analysis::Tag;
```

Comment:
For types, the existing code in `zeek-compat.h` tries to make them available to old Zeek versions by declaring them under their new Zeek name (in contrast to putting them into the `compat` namespace). Can we do that here too?

Reply:
I agree, this would be the right thing to do for long-term maintenance.

It would however be a bigger change, e.g., from a quick look I saw:

- need to provide wrappers for many functions taking or returning `Tag` types so we can convert old `Tag`s into the new `Tag`
- need to provide wrappers for functions of `Tag`, e.g., `AsVal` previously had a different signature

This list could change while I go through the code. I'll try that.

Reply:
Ok, not worth if it's signficant effort.

Reply:
I took a stab at this, and just using `::zeek::Tag` in all versions doesn't really work since it's API changed (e.g., before 4.2 it has no default constructor), so we'd have to come up with our own type anyway. I'd suggest we stay what is in this PR.